chapter3.adoc

```diff
@@ -166,3 +169,102 @@ respective sections in this specification.
+
+[caption="Figure {counter:image}: ", reftext="Figure {image}"]
+[title="MFENCE.SPA instruction"]
+[id=mfence-spa]
+[wavedrom, ,svg]
+....
+{reg: [
+  {bits:  7, name: 'opcode (SYSTEM)'},
+  {bits:  5, name: 'rd (0)'},
+  {bits:  3, name: 'func3 (PRIV)'},
+  {bits:  5, name: 'rs1 (PADDR)'},
+  {bits:  5, name: 'rs2 (SDID'},
+  {bits:  7, name: 'func7 (MFENCE.SPA)'},
+], config:{lanes: 1, hspace:1024}}
+....
+
+The `MFENCE.SPA` fence instruction is used to synchronize updates to supervisor
+domain access-permissions with current execution. `MFENCE.SPA` applies only to
+the caches associated with access-permissions for supervisor domains.
+`MFENCE.SPA` is only valid in M-mode. If operand rs1 is not equal to x0, it
+specifies a single physical address, and if rs2 is not equal to 0, it specifies
+a single SDID. Executing a `MFENCE.SPA` guarantees that any previous stores
+already visible to the current hart are ordered before all implicit reads by
+that hart done for supervisor domain access-permission structures for
+non-M-mode instructions that follow the `MFENCE.SPA`.
+
+When SDID is specified in rs2, bits XLEN-1:SDIDMAX held in rs2 are reserved for
+future standard use. Until their use is specified, they should be zeroed by
+software and ignored by implementations. Also, if SDIDLEN < SDIDMAX, the
+implementation shall ignore bits SDIDMAX-1:SDIDLEN of the value held in rs2.
+
+[NOTE]
+====
+A simpler implementation of MFENCE.SPA may ignore the physical address in rs1,
+and/or the SDID value in rs2, and always perform a global fence for all SDs.
+====
+
+=== M-mode Supervisor Domain Fine-Grain Invalidation Instruction
+
+In some high-performance implementations, a finer-granular invalidation and
+fencing is required that allows for synchronization operations to be more
+efficiently batched. When `Sinval` is implemented with `Smsdid`, the
+`MINVAL.SPA` instruction must be implemented to support such fine-granular
+invalidation of physical memory access-permission caches.
+
+[caption="Figure {counter:image}: ", reftext="Figure {image}"]
+[title="MINVAL.SPA instruction"]
+[id=minval-spa]
+[wavedrom, ,svg]
+....
+{reg: [
+  {bits:  7, name: 'opcode (SYSTEM)'},
+  {bits:  5, name: 'rd (0)'},
+  {bits:  3, name: 'func3 (PRIV)'},
+  {bits:  5, name: 'rs1 (PADDR)'},
+  {bits:  5, name: 'rs2 (SDID'},
+  {bits:  7, name: 'func7 (MINVAL.SPA)'},
+], config:{lanes: 1, hspace:1024}}
+....
+
+`MINVAL.SPA` is only ordered against `SFENCE.W.INVAL` and `SFENCE.INVAL.IR`
+instructions. As part of the update to the SD access-permissions, the RDSM must
+ensure that it uses `SFENCE.W.INVAL` to guarantee that any previous stores to
+structures that hold supervisor domain access-permissions (e.g. `MTT`) are made
+visible before invoking the `MINVAL.SPA`. The RDSM must then use
+`SFENCE.INVAL.IR` to guarantee that all subsequent implicit references to
+supervisor domain access-permission structures (e.g. `MTT`) are ordered to be
```

Comment:
Why not just say - implicit references to MTT instead of as an example.

Reply:
since this section (and inst) is independent of Smmtt